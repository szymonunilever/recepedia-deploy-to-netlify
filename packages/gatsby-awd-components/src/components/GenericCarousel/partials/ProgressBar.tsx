import React from 'react';
import theme from './ProgressBar.module.scss';
import { ProgressBarProps } from './models';

const ProgressBar = ({ percentage = 0 }: ProgressBarProps) => {
  return (
    <div className={theme.progressBar + ' progress-bar'}>
      <progress
        aria-label="progress"
        role="progressbar"
        className={theme.progressBar__content + ' progress-bar__content'}
        style={{ width: `${percentage}%` }}
        value={`${percentage}`}
        max="100"
      >
        {percentage}% complete
      </progress>
    </div>
  );
};

export default ProgressBar;
