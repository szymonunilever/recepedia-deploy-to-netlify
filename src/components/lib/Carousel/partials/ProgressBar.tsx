import React from 'react';
import styles from './ProgressBar.module.scss';
import { ProgressBarProps } from './models';

const ProgressBar = ({ percentage = 0 }: ProgressBarProps) => {
  return (
    <div className={styles.progress_bar}>
      <div
        className={styles.progress_bar_content}
        style={{ width: `${percentage}%` }}
      />
    </div>
  );
};

export default ProgressBar;
