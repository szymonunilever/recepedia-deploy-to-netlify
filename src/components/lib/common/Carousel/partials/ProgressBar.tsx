import React from 'react';
import styles from './ProgressBar.module.scss';
import { ProgressBarProps } from './models';

const ProgressBar = ({ percentage = 0 }: ProgressBarProps) => {
  return (
    <div className={styles['progress-bar']}>
      <div
        className={styles['progress-bar__content']}
        style={{ width: `${percentage}%` }}
      />
    </div>
  );
};

export default ProgressBar;
