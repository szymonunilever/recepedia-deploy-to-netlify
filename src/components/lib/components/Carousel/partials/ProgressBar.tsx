import React from 'react';
import styles from './ProgressBar.module.scss';
import { ProgressBarProps } from './models';

const ProgressBar = ({ percentage = 0 }: ProgressBarProps) => {
  return (
    // <div className={styles['progress-bar'] + ' progress-bar'}>
    <div className={styles.progressBar + ' progress-bar'}>
      <progress
        aria-label="progress"
        role="progressbar"
        className={styles.progressBar__content + ' progress-bar__content'}
        style={{ width: `${percentage}%` }}
        value={`${percentage}`}
        max="100"
      >
        {percentage}% complete
      </progress>
    </div>
  );
};

export default ProgressBar;
