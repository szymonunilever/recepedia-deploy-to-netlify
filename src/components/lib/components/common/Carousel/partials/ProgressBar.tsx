import React from 'react';
import styles from './ProgressBar.module.scss';
import { ProgressBarProps } from './models';

const ProgressBar = ({ percentage = 0 }: ProgressBarProps) => {
  return (
    // <div className={styles['progress-bar'] + ' progress-bar'}>
    <div className={styles.progressBar + ' progress-bar'}>
      <div
        className={styles.progressBar__content + ' progress-bar__content'}
        style={{ width: `${percentage}%` }}
      />
    </div>
  );
};

export default ProgressBar;
