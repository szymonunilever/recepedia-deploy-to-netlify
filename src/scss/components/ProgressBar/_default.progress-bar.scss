.progress-bar{
  background-color: $gray;
  height: 2px;
  width: 102px;
  margin-bottom: 25px;
  @include respondTo(desktop) {
    width: 170px;
  }
  &__content{
    transition: width .2s ease-in;
    height: 2px;
    background:$fish-blue-darker;
  }
}
