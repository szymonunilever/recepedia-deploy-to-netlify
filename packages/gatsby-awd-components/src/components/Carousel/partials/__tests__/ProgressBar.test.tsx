import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import ProgressBar from '../ProgressBar';
import toJson from 'enzyme-to-json';

describe('<ProgressBar />', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<ProgressBar />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
