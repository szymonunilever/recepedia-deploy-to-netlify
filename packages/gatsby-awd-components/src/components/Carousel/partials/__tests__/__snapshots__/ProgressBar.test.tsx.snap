// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> matches the snapshot 1`] = `
<ProgressBar>
  <div
    className="progressBar progress-bar"
  >
    <progress
      aria-label="progress"
      className="progressBar__content progress-bar__content"
      max="100"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
      value="0"
    >
      0
      % complete
    </progress>
  </div>
</ProgressBar>
`;
